include/sys/zpl.h

```diff
@@ -32,12 +32,49 @@
 #include <linux/exportfs.h>
 #include <linux/falloc.h>
 #include <linux/file_compat.h>
+#include <linux/fs.h>
 #include <linux/parser.h>
 #include <linux/task_io_accounting_ops.h>
 #include <linux/vfs_compat.h>
 #include <linux/writeback.h>
 #include <linux/xattr_compat.h>
 
+#ifdef FS_PROJINHERIT_FL
+#define	ZFS_PROJINHERIT_FL	FS_PROJINHERIT_FL
+#else
+#define	ZFS_PROJINHERIT_FL	0x20000000
+#endif
+
+#ifdef FS_IOC_FSGETXATTR
+typedef struct fsxattr zfsxattr_t;
+
+#define	ZFS_IOC_FSGETXATTR	FS_IOC_FSGETXATTR
+#define	ZFS_IOC_FSSETXATTR	FS_IOC_FSSETXATTR
+#else
+struct zfsxattr {
+	__u32		fsx_xflags;	/* xflags field value (get/set) */
+	__u32		fsx_extsize;	/* extsize field value (get/set) */
+	__u32		fsx_nextents;	/* nextents field value (get)   */
+	__u32		fsx_projid;	/* project identifier (get/set) */
```

Comment:
Potentially yes.  Those pad bytes have already been claimed in the upstream kernel so we might as well declare the field to make it clear they're unavailable.